### DISCUSSION TITLE
query_raw and execute_raw

### DISCUSSION THREAD

**Discussion by skbom** [RobertCraigie/prisma-client-py#910] [Q&A]
I'd like to write a write a query bit execute_raw returns an int and query_raw does not seem to accept placeholders. how do I do it?

**This returns an _int_**
`total = await db.execute_raw(
    '''
    SELECT *
    FROM User
    WHERE User.id = ?
    ''',
    'cksca3xm80035f08zjonuubik'
)`

**And I don't know how to safly query, i.e with placeholders with this. Giving the User.id as an extra argument doesn't work**
`posts = await db.query_raw(
    '''
    SELECT *
    FROM User
    WHERE User.id = ?
    '''
)`

**Comment by RobertCraigie** [comment]
What issue did you run into when passing the extra argument to `query_raw`?

**Reply by skbom** [reply]
> query = "select * from assets limit ?;"
> assets = db.query_raw(query, 10)
> print(assets)

Gives:
**_prisma.errors.RawQueryError: ERROR: syntax error at or near ";"_**

Hi Roberts, thank you for the quick response :)
See mabove, I have also tried to pass the 10 as a string.

**Reply by RobertCraigie** [reply]
@skbom try removing the ;

**Reply by skbom** [reply]
@RobertCraigie  without the ; I'm getting the following error:
`prisma.errors.RawQueryError: ERROR: syntax error at end of input`

**Reply by timmedia** [reply]
For anyone landing on this page while debugging: for me this issue was solved by replacing '?' with '$1', as explained in [this comment](https://github.com/RobertCraigie/prisma-client-py/issues/673#issuecomment-1399188989) on issue #673.
